telluric/georaster.py

```diff
@@ -917,6 +917,7 @@ def type_min(dtype):
             dst_array = self.image
 
         dst_array = dst_array.astype(dst_type)
+        dst_array.mask = self.image.mask
```

Comment:
Waiting to hear @drnextgis opinion on this. Not sure why we didn't catch this situation earlier.

Reply:
I'm not sure about this fix: `numpy.ndarray.astype`  should preserve mask so we don't need to assign it manually. Please take a look couple of rows above: https://github.com/satellogic/telluric/blob/0ee8ec26b8201fbb761982d2b1db6cfe7de85a39/telluric/georaster.py#L914

In this part of code we have a deal with mask shrinking:
```python
In [10]: import numpy as np

In [11]: src = np.ma.array([1, 2], mask=[0, 0])

In [12]: src
Out[12]: 
masked_array(data=[1, 2],
             mask=[False, False],
       fill_value=999999)

In [13]: dst = np.float32(1) * src

In [14]: dst
Out[14]: 
masked_array(data=[1., 2.],
             mask=False,
       fill_value=999999)

In [15]: dst.mask is np.ma.nomask
Out[15]: True
```

So this issue is connected with https://github.com/satellogic/telluric/issues/97 because non shrunk masks are assumed here: https://github.com/satellogic/telluric/blob/345f3eaae496a326c64de11e46622a57020c1f8c/telluric/georaster.py#L1096

@Juanlu001 what do you think about such solution?

```python
mask = Image.fromarray(np.rollaxis(np.rollaxis(np.ma.getmaskarray(self.image), 2), 2).astype(np.uint8)[:, :, 0])
```

Reply:
I see that `np.ma.getmaskarray` always gets the full mask:

https://docs.scipy.org/doc/numpy/reference/generated/numpy.ma.getmaskarray.html

So I agree, but maybe we should replace all `XXX.image.mask` with `np.ma.getmaskarray(XXX.image)` everywhere, right?